lib/modules/datasource/cpan/index.ts

```diff
@@ -0,0 +1,130 @@
+import { cache } from '../../../util/cache/package/decorator';
+import type { HttpResponse } from '../../../util/http/types';
+import * as perlVersioning from '../../versioning/perl';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+
+type MetaCpanResult = {
+  hits: {
+    hits: {
+      _source: {
+        module: {
+          name: string;
+          version?: string;
+        }[];
+        distribution: string;
+        date: string;
+        deprecated: boolean;
+        maturity: string;
+        download_url: string;
+      };
+    }[];
+  };
+};
+
+export class CpanDatasource extends Datasource {
+  static readonly id = 'cpan';
+
+  constructor() {
+    super(CpanDatasource.id);
+  }
+
+  override readonly customRegistrySupport = false;
+
+  override readonly defaultRegistryUrls = ['https://fastapi.metacpan.org/'];
+
+  override readonly defaultVersioning = perlVersioning.id;
+
+  @cache({
+    namespace: `datasource-${CpanDatasource.id}`,
+    key: ({ packageName }: GetReleasesConfig) => `${packageName}`,
+  })
+  override async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    // istanbul ignore if
+    if (!registryUrl) {
+      return null;
+    }
+
+    let result: ReleaseResult | null = null;
+    const pkgUrl = `${registryUrl}v1/file/_search`;
+
+    let raw: HttpResponse<MetaCpanResult> | null = null;
+    try {
+      const body = {
+        query: {
+          filtered: {
+            query: { match_all: {} },
+            filter: {
+              and: [
+                { term: { 'module.name': packageName } },
+                { exists: { field: 'module.associated_pod' } },
+              ],
+            },
+          },
+        },
+        _source: [
+          'module.name',
+          'module.version',
+          'distribution',
+          'date',
+          'deprecated',
+          'maturity',
+          'download_url',
```

Comment:
See below, try to get repository url instead
![image](https://user-images.githubusercontent.com/1798109/180128005-cdb43e9f-8876-4f50-8fae-e54b8ced4b6a.png)

Reply:
Looks like that info is not included in the API response.

```json
{
   "hits" : {
      "total" : 169,
      "hits" : [
         {
            "_score" : 1.0,
            "_id" : "p_4tiP86lSIeMm5iaQv4aVAe1WU",
            "_source" : {
               "path" : "lib/Plack.pm",
               "distribution" : "Plack",
               "pod" : "NAME Plack - PSGI reference implementation and utilities DESCRIPTION Plack is a set of PSGI reference server implementations and helper utilities for Web application frameworks, exactly like Ruby's Rack. See PSGI for the PSGI specification. MODULES AND UTILITIES Plack::Server Plack::Server is a namespace for PSGI server implementations. We have Standalone, CGI, FCGI, Apache, AnyEvent, Coro, Danga::Socket and many server environments that you can run PSGI applications on. See Plack::Server how to write your own server implementation. Plack::Loader Plack::Loader is a loader to load one of Plack::Server backends and run PSGI application code reference with it. Plack::Util Plack::Util contains a lot of utility functions for server implementors as well as middleware authors. .psgi files PSGI application is a code reference but it's not easy to pass code reference in the command line or configuration files, so Plack uses a convention that you need a file named \"app.psgi\" or alike, which would be loaded (via perl's core function \"do\") to return the PSGI application code reference. See eg/dot-psgi directory for the example \".psgi\" files. plackup plackup is a command line launcher to run PSGI applications from command line using Plack::Loader to load PSGI backends. It can be used to run standalone servers and FastCGI daemon processes. Other server backends like Apache2 needs a separate configuration but \".psgi\" application file can still be the same. Plack::Middleware PSGI middleware is a PSGI application that wraps existent PSGI application and plays both side of application and servers. From the servers the wrapped code reference still looks like and behaves exactly the same as PSGI applications. Plack::Middleware gives you an easy way to wrap PSGI applications with a clean API, and compatibility with Plack::Builder DSL. Plack::Builder Plack::Builder gives you a DSL that you can enable Middleware in \".psgi\" files to wrap existent PSGI applications. Plack::Request, Plack::Response Plack::Request gives you a nice wrapper API around PSGI $env hash to get headers, cookies and query parameters much like Apache::Request in mod_perl. Plack::Response does the same to construct the response array reference. Plack::Test Plack::Test is an unified interface to test your PSGI application using standard HTTP::Request and HTTP::Response pair with simple callbacks. Plack::Test::Suite Plack::Test::Suite is a test suite to test a new PSGI server backend. AUTHORS Tatsuhiko Miyagawa Yuval Kogman Tokuhiro Matsuno Kazuhiro Osawa Kazuho Oku SEE ALSO PSGI <http://plackperl.org/> LICENSE This library is free software; you can redistribute it and/or modify it under the same terms as Perl itself.",
               "indexed" : true,
               "suggest" : {
                  "payload" : {
                     "doc_name" : "Plack"
                  },
                  "input" : [
                     "Plack"
                  ],
                  "weight" : 995
               },
               "authorized" : true,
               "module" : [
                  {
                     "associated_pod" : "MIYAGAWA/Plack-0.9010/lib/Plack.pm",
                     "indexed" : true,
                     "version_numified" : 0.901,
                     "authorized" : true,
                     "name" : "Plack",
                     "version" : "0.9010"
                  }
               ],
               "version" : "0.9010",
               "abstract" : "PSGI reference implementation and utilities",
               "binary" : false,
               "documentation" : "Plack",
               "date" : "2009-11-12T07:23:02",
               "status" : "backpan",
               "directory" : false,
               "description" : "Plack is a set of PSGI reference server implementations and helper utilities for Web application frameworks, exactly like Ruby's Rack. See PSGI for the PSGI specification.",
               "pod_lines" : [
                  [
                     10,
                     104
                  ]
               ],
               "name" : "Plack.pm",
               "release" : "Plack-0.9010",
               "dist_fav_count" : 226,
               "slop" : 65,
               "download_url" : "https://cpan.metacpan.org/authors/id/M/MI/MIYAGAWA/Plack-0.9010.tar.gz",
               "sloc" : 6,
               "level" : 1,
               "deprecated" : false,
               "stat" : {
                  "mode" : 33188,
                  "gid" : 1009,
                  "mtime" : 1258010515,
                  "uid" : 1009,
                  "size" : 3026
               },
               "author" : "MIYAGAWA",
               "version_numified" : 0.901,
               "mime" : "text/x-script.perl-module",
               "id" : "p_4tiP86lSIeMm5iaQv4aVAe1WU",
               "maturity" : "released"
            },
            "_index" : "cpan_v1_01",
            "_type" : "file"
         }
      ],
      "max_score" : 1.0
   },
   "timed_out" : false,
   "took" : 21,
   "_shards" : {
      "total" : 3,
      "successful" : 3,
      "failed" : 0
   }
}
```

Reply:
It seems that the popup is implemented in MetaCPAN web side: https://github.com/metacpan/metacpan-web/commit/7a0559d5a8c6b5e839802be6b5dcfc5d33508580

Reply:
I understand that [`/release/{distribution}` endpoint](https://github.com/metacpan/metacpan-api/blob/master/docs/API-docs.md#releasedistribution) returns repository info. I'm glad if it's okay to implement such supplementary info in later PRs.

Reply:
sure, we require that info to show changelogs on pull requests